termwiz/src/lineedit/buffer.rs

```diff
@@ -174,15 +174,7 @@ impl LineEditBuffer {
                 position
             }
             Movement::StartOfLine => 0,
-            Movement::EndOfLine => {
-                let mut cursor =
-                    GraphemeCursor::new(self.line.len().saturating_sub(1), self.line.len(), false);
-                if let Ok(Some(pos)) = cursor.next_boundary(&self.line, 0) {
-                    pos
-                } else {
-                    self.cursor
-                }
-            }
+            Movement::EndOfLine => self.line.len(),
```

Comment:
`EndOfLine` is simply set to `self.line.len()`

BTW: `self.cursor` is a **byte index**, and the index(storage) is not related to graphemes(visual), so perhaps calling it `cursor_index` would be clearer.
@Mrreadiness @wez

Reply:
emm... I think the crash might be caused by `self.line.len().saturating_sub(1)`, as the offset is not at the boundary of a Unicode character, which causes `cursor.next_boundary(&self.line, 0)` to panic. Just like `StartOfLine`, `EndOfLine` doesn't need any extra calculation; simply using `self.line.len()` is the most straightforward approach.

Reply:
Hmm, what I think we should do here is:
* Remove this change from this PR, as it has a different thesis from the rest of the PR
* Make a new PR that adds a couple of unit tests around is EndOfLine operation, and fixes the problematic case

Reply:
reverted